packages/features/ee/dsync/components/GroupNameCell.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from "react";
+
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import { Badge, TextField, showToast } from "@calcom/ui";
+import { X, Plus } from "@calcom/ui/components/icon";
+
+const GroupNameCell = (props) => {
+  const [groupNames, setGroupNames] = useState(props.groupNames);
+  const [showTextInput, setShowTextInput] = useState(false);
+  const [newGroupName, setNewGroupName] = useState("");
+  const { t } = useLocale();
+
+  const createMutation = trpc.viewer.dsync.teamGroupMapping.create.useMutation({
+    onSuccess: (data) => {
+      setGroupNames([...groupNames, data.newGroupName]);
```

Comment:
Instead of doing this what you can look at is doing optimistic updates of `trpc.viewer.dsync.teamGroupMapping.get.useQuery({ orgId: 1 });` using tRPC setData